### PR TITLE
Separate the Host service

### PR DESCRIPTION
**Pull request by csillag**
.. from AnnotionUISync.

Earlier, the show/hide frame functionality was implemented as a
background functionality of the AnnotationUISync service,
and it was not directly available for the sidebar code.

This commit separated that functionality into a new service,
and exposes it over a public API.

The test have been updated accordingly, too.

   * * *

This PR is an updated version of #2038.

### CODE REVIEW

**Comment by csillag** [comment]
Quoting from the discussion on the predecessor if this PR:

> I think this looks good. Needs a rebase.

Done.

> If the digest issue doesn't start with this PR let's ignore and remove that comment, though.

The digest issue is fixed by PR #2065.

(The two PRs are independent, but merging this one before that will expose the error on master, so I suggest merging the other one first.)

**Comment by csillag** [comment]
I think this is good to go. (Addressed all comments, tests are green, rebased to master, etc.)

File: h/static/scripts/controllers.coffee [hunk 1]
```diff
@@ -22,13 +22,13 @@ class AppController
   this.$inject = [
     '$controller', '$document', '$location', '$route', '$scope', '$window',
     'auth', 'drafts', 'identity',
-    'permissions', 'streamer', 'annotationUI',
+    'permissions', 'streamer', 'annotationUI', 'host',
```
Comment (gergely-ujvari) [inline h/static/scripts/controllers.coffee]: Unused service host.

File: h/static/scripts/test/annotation-sync-test.coffee [hunk 2]
```diff
@@ -28,11 +28,7 @@ describe 'AnnotationSync', ->
       call: sandbox.stub()
       notify: sandbox.stub()
       onConnect: sandbox.stub()
-      links: [
-        {window: PARENT_WINDOW,    channel: createChannel()}
-        {window: 'ANOTHER_WINDOW', channel: createChannel()}
-        {window: 'THIRD_WINDOW',   channel: createChannel()}
-      ]
+      links: []
```
Comment (gergely-ujvari) [inline h/static/scripts/test/annotation-sync-test.coffee]: Why is this removed here? (I don't see the connection to the PR)

Comment (gergely-ujvari) [inline h/static/scripts/test/annotation-sync-test.coffee]: Oh I see. My bad.

File: h/static/scripts/controllers.coffee (hunk 1 above)
Comment (csillag) [inline h/static/scripts/controllers.coffee]: True, no public API of the host service is called (in this version), but simply instantiating this host enables in to function in the background. So it's not strictly unused. 

(Anyway, the controller will start using the host service in the direct linking work; that's what I put it here.) Can you suggest a different location to place this "unused" import?

Comment (gergely-ujvari) [inline h/static/scripts/controllers.coffee]: We had a talk about this. I suggest putting that part into app.coffee where we configure the other module related services. This way it won't broke if we accidentally we not inject the service to anywhere else

Comment (csillag) [inline h/static/scripts/controllers.coffee]: OK, I moved it..

**Comment by gergely-ujvari** [comment]
Okay, LGTM

**Comment by gergely-ujvari** [comment]
:fireworks:
